File: sample/GNAdSampleMultipleBanner/app/src/main/java/jp/co/geniee/samples/multiplebanner/MemCache.java
package jp.co.geniee.samples.multiplebanner;

import android.graphics.Bitmap;
import android.support.v4.util.LruCache;

public class MemCache {
	private static int cacheSize = 6 * 1024;

	private static LruCache<String, Bitmap> mMemoryCache = new LruCache<String, Bitmap>(cacheSize) {
		@Override
		protected int sizeOf(String key, Bitmap bitmap) {
			return bitmap.getRowBytes() * bitmap.getHeight() / 1024;
		}
 
		protected void entryRemoved(boolean evicted, String key, Bitmap oldValue, Bitmap newValue) {
			if (oldValue != null && !oldValue.isRecycled()) {
			}
			oldValue = null;
		};
	};

	public static Bitmap getImage(String key) {
		Bitmap bitmap = mMemoryCache.get(key);
		if(bitmap != null && bitmap.isRecycled()){
			return null;
		}else {
			return bitmap;
		}
	}

	public static void setImage(String key, Bitmap bitmap) {
		if (!MemCache.hasImage(key) && bitmap != null) {
			mMemoryCache.put(key, bitmap);
		}
	}

	public static boolean hasImage(String key) {
		if (getImage(key) == null) {
			return false;
		} else {
			return true;
		}
	}

	public static void clear() {
		mMemoryCache.evictAll();
	}
}

File: sample/UseMaven/GNAdSamples/kotlinSample/src/main/java/jp/co/geniee/samples/common/MemCache.kt
package jp.co.geniee.samples.common

import android.graphics.Bitmap
import androidx.collection.LruCache

object MemCache {
    private const val cacheSize = 6 * 1024

    private val mMemoryCache = object : androidx.collection.LruCache<String, Bitmap>(cacheSize) {
        override fun sizeOf(key: String, bitmap: Bitmap): Int {
            return bitmap.rowBytes * bitmap.height / 1024
        }

        override fun entryRemoved(evicted: Boolean, key: String, oldValue: Bitmap, newValue: Bitmap?) {
            var oldValue = oldValue
            if (!oldValue.isRecycled) {
            }
        }
    }

    fun getImage(key: String): Bitmap? {
        val bitmap = mMemoryCache.get(key)
        return if (bitmap != null && bitmap.isRecycled) {
            null
        } else {
            bitmap
        }
    }

    fun setImage(key: String, bitmap: Bitmap?) {
        if (!hasImage(key) && bitmap != null) {
            mMemoryCache.put(key, bitmap)
        }
    }

    fun hasImage(key: String): Boolean {
        return getImage(key) != null
    }

    fun clear() {
        mMemoryCache.evictAll()
    }
}
